packages/editor/src/browser/navigation/navigation-location-service.ts

```diff
@@ -233,4 +238,36 @@ ${this.stack.map((location, i) => `${i}: ${JSON.stringify(NavigationLocation.toO
 ----- o -----`;
     }
 
+    /**
+     * Get the last recently closed editor.
+     *
+     * @returns the recently closed editor if it exists.
+     */
+    getLastClosedEditor(): EditorWidget | undefined {
```

Comment:
Minor: the name `getLastClosedEditor` doesn't clearly indicate that the method also mutates the list of previously closed editors. Maybe `dequeueLastClosedEditor` or at least a hint in the JSDoc that this is a queue operation, not just a get operation?

Reply:
After offline discussion with @vince-fugnitto, I decided to change the implementation a bit. 
+ `getLastClosedEditor()` now only returns the last closed editor (as we all expect), without removing anything from the stack.
+ Removing and adding widgets to this stack are all done through `editor-manager` when creating and disposing widgets, respectively.